mlir/include/mlir/Dialect/LLVMIR/ROCDLOps.td

```diff
@@ -192,6 +192,18 @@ def ROCDL_BarrierOp : ROCDL_Op<"barrier"> {
   let assemblyFormat = "attr-dict";
 }
 
+def ROCDL_SetPrioOp : ROCDL_IntrOp<"s.setprio", [], [], [], 0>,
+  Arguments<(ins I16:$priority)> {
+  let results = (outs);
+  let assemblyFormat = "$priority attr-dict";
+}
+
+def ROCDL_SchedBarrier : ROCDL_IntrOp<"sched.barrier", [], [], [], 0>,
+  Arguments<(ins I32:$mask)> {
```

Comment:
Wait, hold on a minute, this is an "ImmArg", it should be an attribute.

Reply:
Any suggestion on how to do that? I mean I tried:
```
def ROCDL_SetPrioOp : ROCDL_IntrOp<"s.setprio", [], [], [], 0>,
  Arguments<(ins I16Attr:$priority)> {
  let results = (outs);
  let assemblyFormat = "attr-dict";
}
```
But this does not work, since the attribute is ignored by `ROCDL_IntrOp`. I tried to manually create a `llvmBuilder`, but the call to `createIntrinsicCall` is not happy with the inputs being non-Values. I also tried `createIntrinsicCallWithRange` without any luck.

Reply:
Nvm, I figured it out! Thanks!